### PR TITLE
Feature/3120 drawer nav item mui style engine

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes # BLUI-3120
Changes proposed in this Pull Request:
Update styles to use Mui-v5 Styled components
Update component to extend sx prop

Test:
There is a showcase branch feature/3120-drawer-nav-item-showcase-example that you can use to test this.
When testing, ensure that there are no breaking changes and that the old API properties still work the same and that you can still style things with inline styles as well as using our classes prop.
Also, test out the new sx prop extension. You can apply styles to the root by using sx directly on the component as well as by targeting nested items by the generated classnames.

### CODE REVIEW

**Comment by emclaug2** [comment]
This comment applies to this PR as well:
https://github.com/brightlayer-ui/react-component-library/pull/432#issuecomment-1126192592

**Comment by bkarambe** [comment]
> This comment applies to this PR as well: [#432 (comment)](https://github.com/brightlayer-ui/react-component-library/pull/432#issuecomment-1126192592)

To test this I have added a branch feature/3120-drawer-nav-item-showcase-example which has DrawerNavItem example added.

**Comment by emclaug2** [comment]
This is not an issue directly for NavItem, but NavGroup is throwing this error in console: 
image

**Comment by daileytj** [comment]
Screen Shot 2022-05-17 at 3 48 13 PM
Is this padding and misaligned icon a result of your changes or extra code in the showcase branch? Either way, let's clean this up

**Comment by bkarambe** [comment]
> Screen Shot 2022-05-17 at 3 48 13 PM Is this padding and misaligned icon a result of your changes or extra code in the showcase branch? Either way, let's clean this up

I am not able to reproduce this, I have checked out feature/3120-drawer-nav-item-showcase-example in showcase

**Comment by emclaug2** [comment]
@bkarambe Try to run the showcase from a fresh clone.  There's some weird caching happening here.  This is becoming an actual problem now since it's impeding our ability to test.  We talked about making a spike story in retro today so we should actually do that.

Here's what I see after a fresh clone.  
Screen Shot 2022-05-20 at 5 27 13 PM

**Comment by daileytj** [comment]
Screen Shot 2022-05-26 at 8 13 28 AM
I'm still seeing this padding issue @bkarambe. Can you try deleting your local copy and pulling this down fresh and running it to reproduce. I would also recommend merging in the latest `release/mui-v5-style-updates` if you haven't already.

**Comment by jeffvg** [comment]
> Screen Shot 2022-05-26 at 8 13 28 AM I'm still seeing this padding issue @bkarambe. Can you try deleting your local copy and pulling this down fresh and running it to reproduce. I would also recommend merging in the latest `release/mui-v5-style-updates` if you haven't already.

I see the same issue on 

image

File: components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx [hunk 1]
```diff
@@ -90,101 +110,112 @@ const calcNestedPadding = (theme: Theme, depth: number): string => {
...
+})<Pick<DrawerNavItemProps, 'depth'>>(({ depth, theme }) => ({
+    '& .MuiListItemButton-root': {
+        // Have to specify both of these. JSS doesn't like to automatically flip the rule when it's calculated from a function
+        paddingLeft: theme.direction === 'rtl' ? theme.spacing(2) : calcNestedPadding(theme, depth),
+        paddingRight: theme.direction === 'ltr' ? theme.spacing(2) : calcNestedPadding(theme, depth),
+    },
+    [`&.${drawerNavItemClasses.nestedTitle}`]: {
+        fontWeight: 400,
+    },
+    [`&.${drawerNavItemClasses.drawerOpen}`]: {},
```
Comment (daileytj) [inline components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx:155]: Why is this empty style rule here?

File: components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx [hunk 2]
```diff
@@ -90,101 +110,112 @@ const calcNestedPadding = (theme: Theme, depth: number): string => {
...
+    cursor: 'inherit',
+    display: 'flex',
+    padding: '1rem',
+    marginRight: '-1rem',
+    alignItems: 'center',
+    justifyContent: 'center',
+    '&$expanded': {
+        transform: 'rotate(180deg)',
+    },
+    [`&.${drawerNavItemClasses.expanded}`]: {},
```
Comment (daileytj) [inline components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx:200]: empty style rule?

File: components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx [hunk 3]
```diff
@@ -1,35 +1,53 @@
...
+        nestedListGroup: ['nestedListGroup'],
+        nestedTitle: ['nestedTitle'],
+        title: ['title'],
+        titleActive: ['titleActive'],
+        ripple: ['ripple'],
+        expanded: ['expanded'],
+        drawerOpen: ['drawerOpen'],
+        noIconTitle: ['noIconTitle'],
+        textSecondary: ['textSecondary'],
+        square: ['square'],
```
Comment (daileytj) [inline components/src/core/Drawer/DrawerNavItem/DrawerNavItem.tsx:45]: Why are these classes being added. The associated styles should be conditionally applied within the `styled` functions. There are plenty examples of this now that you can follow.

**Comment by daileytj** [comment]
- [ ] can't apply a className directly on the navItem
- [ ] chevron classes definition should have never been removed as this is a breaking change

**Comment by emclaug2** [comment]
I was testing the nest nav item style overrides this morning & ran into some trouble applying `sx` styles to the `nestedListGroup` class.  I tried color, backgroundColor, fontSize but there's no changes.

Here's the code snippet I've been using; add this to the `NavigationDrawer.tsx` file:
```
<DrawerNavItem itemID="itemId" title="Drawer Nav Item Test" 
       sx={{ '.BluiDrawerNavItem-nestedListGroup': {
           fontSize: 594
       }}
       }
   >
    <DrawerNavItem itemID="itemId" title="Drawer Nav Item Test" 
       sx={{ '.BluiDrawerNavItem-nestedListGroup': {
            fontSize: 893
        }}}  />
</DrawerNavItem>
```

**Comment by jeffvg** [comment]
> I was testing the nest nav item style overrides this morning & ran into some trouble applying `sx` styles to the `nestedListGroup` class. I tried color, backgroundColor, fontSize but there's no changes.
> 
> Here's the code snippet I've been using; add this to the `NavigationDrawer.tsx` file:
> 
> ```
> <DrawerNavItem itemID="itemId" title="Drawer Nav Item Test" 
>        sx={{ '.BluiDrawerNavItem-nestedListGroup': {
>            fontSize: 594
>        }}
>        }
>    >
>     <DrawerNavItem itemID="itemId" title="Drawer Nav Item Test" 
>        sx={{ '.BluiDrawerNavItem-nestedListGroup': {
>             fontSize: 893
>         }}}  />
> </DrawerNavItem>
> ```

I have the same, can't use style or className as well

**Comment by daileytj** [comment]
@jeffvg I don't think the example you shared there is a `nestedListGroup`, it's just a nested list item. I'll double check this stuff though. It is quite hard to test the nuances of this

**Comment by daileytj** [comment]
@emclaug2 @jeffvg I'm going to go through this again. I need to verify on dev how these are currently working.

**Comment by daileytj** [comment]
`style` and `className` props don't work on the dev branch for `DrawerNavItem` so if we want those, I propose we do it in another issue. I'm seeing the same updates when using the `classes` prop so that's good. It seems as though the nestedTitle overrides weren't working previously so I'll chalk that up to an ongoing bug if we're expecting that to work differently, but the sx override using the global class for `BluiDrawerNavItem-nestedListGroup` is legit not working. I'll see what I can do about this.

**Review by daileytj — APPROVED** [review]

**Pull request merged**